### PR TITLE
Sink binding docs

### LINKED ISSUE #3510: [1.0] Fully document sinkbindings.sources.knative.dev
@n3wscott commented on [Mon Feb 22 2021](https://github.com/knative/eventing/issues/4918)

Related to https://github.com/knative/docs/issues/3268
The following work needs to be validated for sinkbindings.sources.knative.dev to be ready for 1.0:
- [ ] Getting Started demo using the component.
- [ ] Sample minimum of component
- [ ] Sample complete resource of component
- [ ] Fully documented features and options, including
    - [ ] Explanation of resource.
    - [ ] All spec fields
    - [ ] All status fields
    - [ ] Any magic labels or annotations that apply.
    - [ ] With kubectl cli
    - [ ] With kn cli

---

@snneji commented on [Tue Apr 27 2021](https://github.com/knative/eventing/issues/4918#issuecomment-827616695)

/assign

### PR DESCRIPTION
**Pull request by snneji**
Closes #3510

Worked with @n3wscott on this so the docs should now provide:
- Getting Started demo using the component.
- Sample minimum of component
- Explanation of resource.
- All spec fields
- With kubectl cli
- With kn cli

### CODE REVIEW

File: docs/eventing/sources/sinkbinding/_index.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+title: "SinkBinding"
+weight: 01
+type: "docs"
+aliases:
+  - /docs/eventing/samples/sinkbinding/index
+  - /docs/eventing/samples/sinkbinding/README
+  - /docs/eventing/sources/sinkbinding.md
```
Comment (abrennan89) [inline docs/eventing/sources/sinkbinding/_index.md:8]:
```suggestion
  - /docs/eventing/samples/sinkbinding/
  - /docs/eventing/sources/sinkbinding
```

I think this would also work but maybe check with @RichieEscarez

File: docs/eventing/sources/sinkbinding/_index.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+---
+
+![API version v1](https://img.shields.io/badge/API_Version-v1-red?style=flat-square)
+
+The `SinkBinding` custom object supports decoupling event production from
+delivery addressing.
```
Comment (abrennan89) [inline docs/eventing/sources/sinkbinding/_index.md:14]: I'd go with either "object" or "custom resource" instead of _custom object_

File: docs/eventing/sources/sinkbinding/_index.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+You can use a SinkBinding to direct a subject to an event sink. The _subject_
+is a Kubernetes resource that embeds a `PodSpec` template and produces events.
+The _event sink_ is an addressable Kubernetes object that can receive events.
```
Comment (abrennan89) [inline docs/eventing/sources/sinkbinding/_index.md:18]:
```suggestion
You can use sink binding to direct a subject to an event sink. A _subject_ is a Kubernetes resource that embeds a `PodSpec` template and produces events. An _event sink_ is an addressable Kubernetes object that can receive events.
```

File: docs/eventing/sources/sinkbinding/_index.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+To create new subjects you can use any of the compute objects that Kubernetes
+makes available such as:
+
+- `Deployment`
+- `Job`
+- `DaemonSet`
+- `StatefulSet`
```
Comment (abrennan89) [inline docs/eventing/sources/sinkbinding/_index.md:26]:
```suggestion
To create new subjects you can use any addressable Kubernetes resource, such as:

- `Deployment`
- `Job`
- `DaemonSet`
- `StatefulSet`
```

File: docs/eventing/sources/sinkbinding/_index.md (hunk 1 above)
Comment (RichieEscarez) [reply docs/eventing/sources/sinkbinding/_index.md:8]: Yes, we want the site URLs here not the source paths -> exclude the file type suffix

Comment (RichieEscarez) [reply docs/eventing/sources/sinkbinding/_index.md:8]: Also, these aliases work on staging, so if you are unsure, you can test them in the "Preview" at https://app.netlify.com/sites/knative/deploys

File: docs/eventing/sources/sinkbinding/_index.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+You can also create new subjects that use Knative abstractions, such as
+`Service` or `Configuration` objects.
```
Comment (abrennan89) [inline docs/eventing/sources/sinkbinding/_index.md:29]: These are still custom resources, just for Knative rather than Kubernetes, but they're all addressable Kubernetes objects, so I'd maybe reword this a bit to make it clearer. I'm actually not really sure it's important to call all of this out because there are a ton of things that can be used for sinks.

Out of curiosity, why use the word _subject_ above instead of just _sink_ consistently?

File: docs/eventing/sources/sinkbinding/_index.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+The SinkBinding injects environment variables into the `PodTemplateSpec` of the
```
Comment (abrennan89) [inline docs/eventing/sources/sinkbinding/_index.md:31]:
```suggestion
Sink binding injects environment variables into the `PodTemplateSpec` of the
```

Comment (abrennan89) [reply docs/eventing/sources/sinkbinding/_index.md:31]: Only refer to SinkBinding if we mean the literal object / custom resource here, it seems more conceptual though? Just a suggestion

File: docs/eventing/sources/sinkbinding/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,298 @@
+---
+title: "Create a SinkBinding"
+weight: 02
+type: "docs"
+---
+
+![API version v1](https://img.shields.io/badge/API_Version-v1-red?style=flat-square)
+
+This topic describes how to create a SinkBinding and connect it to a subject in your cluster.
```
Comment (abrennan89) [inline docs/eventing/sources/sinkbinding/getting-started.md:9]: I'd prefer to see it used as "sink binding" or "SinkBinding object"

**Review by abrennan89 — COMMENTED** [review]
some suggestions

File: docs/eventing/sources/sinkbinding/_index.md (hunk 2 above)
Comment (snneji) [reply docs/eventing/sources/sinkbinding/_index.md:13]: Changed to "object"

File: docs/eventing/sources/sinkbinding/_index.md (hunk 6 above)
Comment (snneji) [reply docs/eventing/sources/sinkbinding/_index.md:27]: I think this is referring to what the literal object does, right? I'll update to "SinkBinding object".

File: docs/eventing/sources/sinkbinding/_index.md (hunk 5 above)
Comment (snneji) [reply docs/eventing/sources/sinkbinding/_index.md:25]: About the list:
I'll remove this list as it is also mentioned on the "Create a SinkBinding" page.

About the word subject:
I worked with an engineer on this who said that "subject" was a better word than "source". I thought it was reasonable since the command to create the SinkBinding uses the flags `--subject` and a `--sink`, as in:

```
kn source binding create <name> \
  --namespace <namespace> \
  --subject "<subject>" \
  --sink <event-sink>
```

Do you disagree?

**Comment by snneji** [comment]
Closing as this PR has been migrated to mkdocs branch.
